shared/params.go

```diff
@@ -73,6 +73,29 @@ type ProductSpec struct {
 	Labels mapset.Set
 }
 
+// Matches returns whether the spec matches the given run.
+func (p ProductSpec) Matches(run TestRun) bool {
+	if run.BrowserName != p.BrowserName {
+		return false
+	}
+	if !IsLatest(p.Revision) && p.Revision != run.Revision {
+		return false
+	}
+	if p.Labels != nil && p.Labels.Cardinality() > 0 {
+		runLabels := run.LabelsSet()
+		if runLabels.Intersect(p.Labels).Cardinality() < p.Labels.Cardinality() {
+			return false
+		}
+	}
+	if p.BrowserVersion != "" {
+		// Make "6" not match "60.123" by adding trailing dots to both.
+		if strings.Index(run.BrowserVersion+".", p.BrowserVersion+".") != 0 {
```

Comment:
I think `HasPrefix` would be slightly faster and clearer.

```go
if runVersion := run.BrowserVersion+"."; !runVersion.HasPrefix(p.BrowserVersion+".") {
```

Reply:
I think it's `strings.HasPrefix(s, prefix)`, but, done nonetheless.

Reply:
Oops. Apparently I couldn't wrap my head around Python and Go.